docs/adr/adr_004_server_side_filtering.md

```diff
@@ -0,0 +1,142 @@
+- **Cost of implementation**: Low → High: how much efforts does it represent
+- **Cost of operation**: Low → High: how much does the solution cost to run
+
+## Considered Options
+
+1. [Option 0 - Do nothing](#option-0---do-nothing)
+1. [Option 1 - Split into several collections](#option-1---split-into-several-collections)
+1. [Option 2 - Read-only mirrors](#option-2---read-only-mirrors)
+1. [Option 3 - Implement notion of datasets](#option-3---implement-notion-of-datasets)
+1. [Option 4 - Implement dynamic signatures](#option-4---implement-dynamic-signatures)
+
+## Decision Outcome
+
+Chosen option: if it fits the segmentation requirements, we would pick *Option 2*. We already have all the pieces in place, it is very low tech and low cost, while still providing a good user experiences for editors and publication of data.
+
+## Pros and Cons of the Options
+
+### Option 0 - Do nothing
+
+No change is made. All clients download everything and filter locally using JEXL targeting.
+
+- **Complexity**: N/A
+- **User experience**: Good, users edit a single collection.
+- **Cost of implementation**: N/A
+- **Cost of operation**: High. All clients download everything, resulting in higher bandwidth costs.
+
+
+### Option 1 - Split into several collections
+
+The single collections is split into several ones, and the client is able to determine which collection it should pull from (eg. one per region).
+
+If a record is required in different subsets, it is duplicated in each collection.
+
+- **Complexity**: N/A
+- **User experience**: Bad, even if a script could automate the publication of records into several collections, editors don't have a single source of truth for the content, leading to duplicated actions and data.
+- **Cost of implementation**: Low. Creating collections is cheap.
+- **Cost of operation**: Low. Clients download only the subset of data, saving bandwidth.
+
+
+### Option 2 - Read-only mirrors
+
+With this solution, the main collection is maintained, and several "*side collections*" are created. As with *Option 1*, the clients are able to pick which collection to pull from.
+
+Unlike *Option 1*, the additional collections are read-only. Editors publish data in the main collection, and a scheduled job will copy the records at regular intervals to the side collections using filters.
+
+The [`backport_records` cronjob](https://github.com/mozilla-services/remote-settings-lambdas?tab=readme-ov-file#backport_records) copies records from one collection to another and signs it. It can take querystring filters as parameters in order to only copy of subset of the source.
+
+For example, if records have a `regions` field:
+
+```
+{
+    "id": "shop-A",
+    "regions": ["europe", "asia", "north-america"],
+},
+{
+    "id": "shop-B",
+    "regions": ["europe", "africa"],
+}
+```
+
+Then different collections can be populated using this field in [a querystring filter](https://docs.kinto-storage.org/en/latest/api/1.x/filtering.html#comparison):
```

Comment:
Question: is it possible to support multiple dimensions for side collections? For example:
```
{
    "id": "ship-A",
    "regions": ["europe", "asia", "north-america"],
    "locales": ["en", "fr", "de", "es"],
}

* `shops-europe-en`: `?contains_regions=["europe"]&contains_locales=["en"]`
* `shops-enrope-fr`: `?contains_regions=["europe"]&contains_locales=["fr"]`
....
```

Reply:
Yes it is possible, will update the doc :) 

All conditions are a conjunct ("AND"), and we don't have a way to do disjunctions though ("OR") ([issue opened since 2016](https://github.com/Kinto/kinto/issues/639) 🤣)

Reply:
Very nice! Conjunctions are all we need :)